firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -393,32 +419,42 @@ AppDistributionReleaseInternal getCachedNewRelease() {
     if (newRelease.getReleaseNotes() != null && !newRelease.getReleaseNotes().isEmpty()) {
       message.append(String.format("\n\nRelease notes: %s", newRelease.getReleaseNotes()));
     }
-    updateDialog.setMessage(message);
+    updateConfirmationDialog.setMessage(message);
 
-    updateDialog.setButton(
+    updateConfirmationDialog.setButton(
         AlertDialog.BUTTON_POSITIVE,
         context.getString(R.string.update_yes_button),
         (dialogInterface, i) -> showUpdateDialogTask.setResult(null));
 
-    updateDialog.setButton(
+    updateConfirmationDialog.setButton(
         AlertDialog.BUTTON_NEGATIVE,
         context.getString(R.string.update_no_button),
-        (dialogInterface, i) ->
-            showUpdateDialogTask.setException(
-                new FirebaseAppDistributionException(
-                    ErrorMessages.UPDATE_CANCELED, Status.INSTALLATION_CANCELED)));
+        (dialogInterface, i) -> updateConfirmationDialogClosedCallback(showUpdateDialogTask));
 
-    updateDialog.setOnCancelListener(
-        dialogInterface ->
-            showUpdateDialogTask.setException(
-                new FirebaseAppDistributionException(
-                    ErrorMessages.UPDATE_CANCELED, Status.INSTALLATION_CANCELED)));
+    updateConfirmationDialog.setOnCancelListener(
+        dialogInterface -> updateConfirmationDialogClosedCallback(showUpdateDialogTask));
 
-    updateDialog.show();
+    updateConfirmationDialog.show();
 
     return showUpdateDialogTask.getTask();
   }
 
+  private void setRemakeDialogsToFalse() {
+    synchronized (updateIfNewReleaseTaskLock) {
+      remakeSignInConfirmationDialog = false;
+      remakeUpdateConfirmationDialog = false;
+    }
+  }
+
+  private void updateConfirmationDialogClosedCallback(TaskCompletionSource showUpdateDialogTask) {
+    synchronized (updateIfNewReleaseTaskLock) {
+      remakeUpdateConfirmationDialog = false;
```

Comment:
We don't need this anymore, right? I'd just drop it and inline this method.

Reply:
We don't have this on signIn but it might be nice to have since it does set the remake variable to false when the dialog is cancelled. That may be unnecessary but it makes sense to set it after that since it does not need to be remade at that point.  Curious on your thoughts about that.